src/common/gui/SurgeGUIEditor.cpp

```diff
@@ -5833,39 +5835,40 @@ VSTGUI::CCommandMenuItem* SurgeGUIEditor::addCallbackMenu(VSTGUI::COptionMenu* t
 #if TARGET_VST3
 Steinberg::tresult PLUGIN_API SurgeGUIEditor::onSize(Steinberg::ViewRect* newSize)
 {
-   Steinberg::ViewRect currentSize = *newSize;
+   float width = newSize->getWidth();
+   float height = newSize->getHeight();
 
    // Cubase sets wrong newSize when reopening editor. See #1460
    auto scaleFactorOnClose = synth->storage.getPatch().dawExtraState.editor.scaleFactorOnClose;
    if (scaleFactorOnClose > 1)
    {
-      currentSize.bottom *= scaleFactorOnClose;
-      currentSize.right *= scaleFactorOnClose;
+	  width *= scaleFactorOnClose;
+	  height *= scaleFactorOnClose;
       synth->storage.getPatch().dawExtraState.editor.scaleFactorOnClose = 1; // Don't rescale infinitely
    }
 
-   float izfx = currentSize.getWidth() * 1.0 / getWindowSizeX() * 100.0;
-   float izfy = currentSize.getHeight() * 1.0 / getWindowSizeY() * 100.0;
-   float izf = std::min(izfx, izfy);
-   izf = std::max(izf, 1.0f*minimumZoom);
+   // resolve current zoomFactor
+   float izfx = ceil(width / getWindowSizeX() * 100.0);
+   float izfy = ceil(height / getWindowSizeY() * 100.0);
+   float currentZoomFactor = std::min(izfx, izfy);
+   currentZoomFactor = std::max(currentZoomFactor, 1.0f * minimumZoom);
 
-   int fzf = (int)izf;
    // Don't allow me to set a zoom which will pop a dialog from this drag event. See #1212
-   if( ! doesZoomFitToScreen((int)izf, fzf) )
+   float correctedZoomFactor;
+   if( ! doesZoomFitToScreen(currentZoomFactor, correctedZoomFactor) )
    {
-      izf = fzf;
+	   currentZoomFactor = correctedZoomFactor;
    }
 
-   /*
-   ** If the implied zoom factor changes my size by more than a pixel, resize
-   */
-   auto zfdx = std::fabs( (izf - zoomFactor) * getWindowSizeX() ) / 100.0;
-   auto zfdy = std::fabs( (izf - zoomFactor) * getWindowSizeY() ) / 100.0;
-   auto zfd = std::max( zfdx, zfdy );
+   // If the resolved currentZoomFactor changes size by more than a pixel, store new size
+   auto zfdx = std::fabs( (currentZoomFactor - zoomFactor) * getWindowSizeX() ) / 100.0;
+   auto zfdy = std::fabs( (currentZoomFactor - zoomFactor) * getWindowSizeY() ) / 100.0;
+   bool windowDragResize = std::max( zfdx, zfdy ) > 1;
 
-   if( zfd > 1 )
+   if(windowDragResize)
    {
-      setZoomFactor( izf );
+	  printf("SurgeGUIEditor::onSize setZoomFactor %f\n", currentZoomFactor);
```

Comment:
Probably don't want to leave this printf in here obvs

Reply:
whoops